Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -57,9 +58,28 @@ bool LogFileReader::readAll(const QString& fileName)
  * that the conversion from string to double is sucessful.*/
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
+    // check for exact format "hh:mm:ss.zzz, voltage, current"
+    // time must be hh:mm:ss.zzz
+    static const QString regexHour = "([0-1]\\d|2[0-3])";
+    static const QString regexMinSec = "[0-5]\\d";
+    static const QString regexMSec = "\\d+";
+    static const QString regexTime = regexHour + ":" + regexMinSec + ":" + regexMinSec + "\\." + regexMSec;
+
+    // voltage must be a positive number, current can be any number
+    static const QString regexVoltage = "\\d+.\\d+";
+    static const QString regexCurrent = "(-\\d|\\d)+\\.\\d+";
+
+    // putting it all together
+    static const QString regexItem = QString("(^%1, %2, %3$)").arg(regexTime, regexVoltage, regexCurrent);
+    static const QRegExp rxItem(regexItem);
```

Comment:
Interesting approach to make everything static so that it would only compile once. I think it would've been better to do it in the constructor and add a member variable.

Reply:
I was looking over a lot of arguments for both ways. I think it's better to put in the function to limit the scope and also see that it's only used in that function.

Reply:
Fair :)